src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -315,12 +315,28 @@ public static function createFromGlobals()
         }
 
         $request = self::createRequestFromFactory($_GET, $_POST, array(), $_COOKIE, $_FILES, $server);
+        $normalizedMethod = strtoupper($request->server->get('REQUEST_METHOD', 'GET'));
 
         if (0 === strpos($request->headers->get('CONTENT_TYPE'), 'application/x-www-form-urlencoded')
-            && in_array(strtoupper($request->server->get('REQUEST_METHOD', 'GET')), array('PUT', 'DELETE', 'PATCH'))
+            && in_array($normalizedMethod, array('PUT', 'DELETE', 'PATCH'))
         ) {
             parse_str($request->getContent(), $data);
             $request->request = new ParameterBag($data);
+        } elseif (0 === strpos($request->headers->get('CONTENT_TYPE'), 'application/json')
+            && in_array($normalizedMethod, array('POST', 'PUT', 'DELETE', 'PATCH'))
+        ) {
+            $content = trim($request->getContent());
+            $data = array();
+
+            // only JSON objects are allowed
+            if ('{' === $content[0]) {
+                $result = json_decode($content, true);
```

Comment:
I was looking mainly on the ParameterBag class, where the entry key is a string and the entry value is a mixed value, that way the ParameterBag is closer to be a map collection than a list collection, so such request with an array notation `{ [{"name": "foo"}, {"name": "bar"}] }` does not seem to be good. 
This feature mainly aims for a form based data where everything can be easily presented as a single model (as an aggregate even) and it plays well with the Form component.